File: backend/api/src/main/java/com/coinai/api/category/dto/response/CategoryResponse.java
package com.coinai.api.category.dto.response;

import com.coinai.api.movement.MovementType;
import lombok.Builder;
import lombok.Getter;

import java.time.LocalDateTime;
import java.util.UUID;

@Getter
@Builder
public class CategoryResponse {

    private UUID id;

    private String name;

    private String icon;

    private String color;

    private MovementType movementType;

    private boolean isDefault;

    private LocalDateTime createdAt;

}

File: mobile/lib/features/category/data/models/category_response.dart
class CategoryResponse {
  final String id;
  final String name;
  final String? icon;
  final String? color;
  final String movementType;
  final String origin;

  const CategoryResponse({
    required this.id,
    required this.name,
    this.icon,
    this.color,
    required this.movementType,
    required this.origin,
  });

  factory CategoryResponse.fromJson(
    Map<String, dynamic> json,
  ) {
    return CategoryResponse(
      id: json['id'] as String,
      name: json['name'] as String,
      icon: json['icon'] as String?,
      color: json['color'] as String?,
      movementType: json['movementType'] as String,
      origin: json['origin'] as String,
    );
  }
}